tests/integration/test_apply_checks.py

```diff
@@ -6900,6 +6909,129 @@ def test_compare_datasets_check(ws, spark, set_utc_timezone):
     assert_df_equality(checked.sort(pk_columns), expected.sort(pk_columns), ignore_nullable=True)
 
 
+def _run_llm_pk_detection_test(ws, src_table):
+    """Helper function to run LLM primary key detection test logic."""
+    if not HAS_PROFILER:
+        pytest.skip("DQProfiler not available")
+
+    profiler = DQProfiler(ws)
+
+    # Detect primary keys using actual LLM functionality
+    pk_detection_result = profiler.detect_primary_keys_with_llm(
+        src_table,
+        options={
+            "enable_llm_pk_detection": True,
+            "llm_pk_detection_endpoint": "databricks-meta-llama-3-1-8b-instruct",
+            "llm_pk_validate_duplicates": False,  # Skip duplicate validation for test speed
+        },
+        llm=True,
+    )
+
+    # Skip test if LLM detection failed (e.g., endpoint not available)
+    if pk_detection_result is None or not pk_detection_result.get("success", False):
+        pytest.skip("LLM-based primary key detection not available or failed")
+
+    return pk_detection_result
+
+
+def _create_llm_dataset_check(pk_detection_result, detected_pk_columns, ref_table):
+    """Helper function to create dataset check with LLM-detected primary key."""
+    return DQDatasetRule(
+        name="llm_detected_pk_compare_datasets",
+        criticality="error",
+        check_func=check_funcs.compare_datasets,
+        columns=detected_pk_columns,
+        filter="customer_id != 1002",  # Filter out the middle record
+        check_func_kwargs={"ref_columns": detected_pk_columns, "ref_table": ref_table},
+        user_metadata={
+            "test_tag": "llm_integration",
+            "llm_detected_pk": True,
+            "pk_detection_confidence": pk_detection_result["confidence"],
+            "pk_detection_reasoning": pk_detection_result["reasoning"],
+        },
+    )
+
+
+def _verify_llm_test_results(checked, detected_pk_columns):
+    """Helper function to verify LLM test results."""
+    # Verify that the check was applied and used the LLM-detected primary key
+    assert checked is not None
+    assert "dq_issues" in checked.columns
+    assert "dq_validations" in checked.columns
+
+    # Check that the metadata includes LLM detection information
+    issues_rows = checked.filter(checked.dq_issues.isNotNull()).collect()
+    if len(issues_rows) > 0:
+        issue_data = issues_rows[0]["dq_issues"][0]
+        assert issue_data["name"] == "llm_detected_pk_compare_datasets"
+        assert issue_data["user_metadata"]["llm_detected_pk"] is True
+        assert "pk_detection_confidence" in issue_data["user_metadata"]
+        assert "pk_detection_reasoning" in issue_data["user_metadata"]
+        # Verify the columns used match what LLM detected
+        assert issue_data["columns"] == detected_pk_columns
+
+
+def test_compare_datasets_check_with_llm_pk_detection(ws, spark, set_utc_timezone):
```

Comment:
i don't really understand the purpose of this test. There is no need to test compare datasets if there are no changes in it.